pkg/resource/replication_group/testdata/replication_group/cr/rg_cmd_before_engine_version_upgrade.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: elasticache.services.k8s.aws/v1alpha1
+kind: ReplicationGroup
+# omitted metadata
+spec:
+  cacheNodeType: cache.t3.micro
+  engine: redis
+  engineVersion: 5.0.6 # new config has been applied; this is the new engine version but no action has been taken yet
+  numNodeGroups: 1
+  replicasPerNodeGroup: 1
+  replicationGroupDescription: cluster-mode disabled RG
+  replicationGroupID: rg-cmd
+status:
+  ackResourceMetadata:
+    arn: arn:aws:elasticache:us-east-1:012345678910:replicationgroup:rg-cmd
+    ownerAccountID: ""
+  allowedScaleUpModifications:
+    - cache.m3.2xlarge
+    - cache.m3.large
+    - cache.m3.medium
+    - cache.m3.xlarge
+    - cache.m4.10xlarge
+    - cache.m4.2xlarge
+    - cache.m4.4xlarge
+    - cache.m4.large
+    - cache.m4.xlarge
+    - cache.m5.12xlarge
+    - cache.m5.24xlarge
+    - cache.m5.2xlarge
+    - cache.m5.4xlarge
+    - cache.m5.large
+    - cache.m5.xlarge
+    - cache.r3.2xlarge
+    - cache.r3.4xlarge
+    - cache.r3.8xlarge
+    - cache.r3.large
+    - cache.r3.xlarge
+    - cache.r4.16xlarge
+    - cache.r4.2xlarge
+    - cache.r4.4xlarge
+    - cache.r4.8xlarge
+    - cache.r4.large
+    - cache.r4.xlarge
+    - cache.r5.12xlarge
+    - cache.r5.24xlarge
+    - cache.r5.2xlarge
+    - cache.r5.4xlarge
+    - cache.r5.large
+    - cache.r5.xlarge
+    - cache.t2.medium
+    - cache.t2.micro
+    - cache.t2.small
+    - cache.t3.medium
+    - cache.t3.small
+  authTokenEnabled: false
+  automaticFailover: disabled
+  clusterEnabled: false
+  conditions:
+    - status: "True"
+      type: ACK.ResourceSynced
+  description: cluster-mode disabled RG
+  events:
+    - date: "2021-04-13T19:07:05Z"
+      message: Replication group rg-cmd created
+  globalReplicationGroupInfo: {}
+  memberClusters:
+    - rg-cmd-001
+    - rg-cmd-002
+  multiAZ: disabled
+  nodeGroups:
+    - nodeGroupID: "0001"
+      nodeGroupMembers:
+        - cacheClusterID: rg-cmd-001
```

Comment:
Just curious, but if cache mode is disabled, how are the RG node groups related to a cache cluster?

Reply:
@jaypipes Sorry I'm not quite following, did you mean cluster mode? 
"Cluster mode disabled (CMD)" = 1 node group
"Cluster mode enabled (CME)" = 2-500 node groups (but can be 1 node group as long as it's created with a `.cluster.on` parameter group...)

Where each node group has 1 primary node and 0-5 replica nodes, and nodes are synonymous with "cache clusters", at least for Redis (which is confusing because the entire replication group or one of its node groups can be considered a "cache cluster" in some sense).  @nmvk please correct me if I made any errors

I realize this is just clarifying terminology and may not answer your question, but wanted to make sure we're on the same page since I didn't quite understand the question.

Reply:
@echen-98 it's just super confusing ElastiCache API, frankly...

If something is "cluster mode disabled", it is a surprising API user experience to see something called "cacheClusterID" in that something's attributes.

Reply:
I think part of the confusion with the term "cache cluster" is that with Redis, it's just a single cache node, but with Memcached, it can be 1-20 nodes. But yeah things could probably be named more clearly.